libsel4vm/src/arch/x86/guest_x86_context.c

```diff
@@ -4,6 +4,13 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+/* Under release mode, this file will get built using -O3. Howerver on gcc 8.4,
```

Comment:
Is this a problem specific to GCC 8.4 or does it happen with newer versions and clang also? Such issues leave an odd feeling that there is an issue with the code (maybe alignment?) that I consider worth investigating. However, it should not be seen as a blocker for merging this as it temporarily fixes a problem.

I have created https://github.com/seL4/seL4_projects_libs/issues/75 to track this.